Sources/SwiftSyntaxMacroExpansion/MacroSystem.swift

```diff
@@ -131,13 +124,29 @@ private func expandFreestandingExpr(
     return nil
   }
 
-  let indentedSource =
-    expanded
-    .indented(by: node.indentationOfFirstLine)
-    .wrappingInTrivia(from: node)
+  let indentedSource = adjustIndentationOfFreestandingMacro(expandedCode: expanded, node: node)
+
   return "\(raw: indentedSource)"
 }
 
+/// Adds the appropriate indentation on expanded code even if it's multi line.
+/// Makes sure original macro expression's trivia is maintained by adding it to expanded code.
+private func adjustIndentationOfFreestandingMacro(expandedCode: String, node: some FreestandingMacroExpansionSyntax) -> String {
+  let indentationOfFirstLine = node.indentationOfFirstLine
+
+  var indentedSource =
+    expandedCode
+    .indented(by: indentationOfFirstLine)
+    .wrappingInTrivia(from: node)
+
+  // if the experssion started in middle of the line, then remove indentation of the first line
+  if !node.leadingTrivia.contains(where: \.isNewline) {
+    indentedSource.removeFirst(indentationOfFirstLine.sourceLength.utf8Length)
+  }
```

Comment:
I think this might be incorrect if `wrappingInTrivia` adds something (like a comment) because then the first `indentationOfFirstLine.sourceLength.utf8Length` characters in `indentedSource` aren’t actually the indentation but whatever `wrappingInTrivia` added.

So, this probably needs to be done before calling `wrappingInTrivia`.

Reply:
Thanks for noticing and letting me know. I have updated the logic and added some unit tests where `wrappingInTrivia` will have comment trivia.

Please review again.